app/main.py

```diff
@@ -2,5 +1,14 @@
+from typing import Callable, Any
 
 
-def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+def cache(func: callable) -> Callable:
+    cache_result = {}
+
+    def function_cache(*args) -> Any:
+        if args not in cache_result:
+            cache_result[*args] = func(*args)
```

Comment:
The syntax `cache_result[*args]` is incorrect for dictionary key assignment. It should be `cache_result[args]` since `args` is already a tuple.